File: web/src/lib/format.ts
/** Display formatting shared by the site. Prices arrive as integer paise; dates as ISO `YYYY-MM-DD`. */

const rupees = new Intl.NumberFormat('en-IN', {
  style: 'currency',
  currency: 'INR',
  maximumFractionDigits: 0,
});

/** `14_499_00` → `₹14,499` (no space after the symbol on any ICU build). */
export const inr = (paise: number) => rupees.format(Math.round(paise / 100)).replace(/\s/g, '');

const WEEKDAYS = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
export const MONTHS = [
  'Jan',
  'Feb',
  'Mar',
  'Apr',
  'May',
  'Jun',
  'Jul',
  'Aug',
  'Sep',
  'Oct',
  'Nov',
  'Dec',
];

/** Parse an ISO date as a calendar day (UTC), so the server's timezone never shifts it. */
const day = (iso: string) => new Date(`${iso}T00:00:00Z`);

/** `2026-11-20` → `Fri 20 Nov 2026` */
export function formatDate(iso: string) {
  const d = day(iso);
  return `${WEEKDAYS[d.getUTCDay()]} ${d.getUTCDate()} ${MONTHS[d.getUTCMonth()]} ${d.getUTCFullYear()}`;
}

/** `2026-12-18` → `18 Dec` */
export function shortDate(iso: string) {
  const d = day(iso);
  return `${d.getUTCDate()} ${MONTHS[d.getUTCMonth()]}`;
}

export const duration = (nights: number, days: number) => `${nights}N / ${days}D`;

export function mealsLabel(meals: { breakfast: boolean; lunch: boolean; dinner: boolean }) {
  const names = [
    meals.breakfast && 'Breakfast',
    meals.lunch && 'Lunch',
    meals.dinner && 'Dinner',
  ].filter(Boolean);
  return names.length ? names.join(' · ') : 'No meals';
}

/**
 * `[11, 12, 1, 2]` → `Nov – Feb`. Runs may wrap the year end; several runs join with ` · `
 * (`Mar – Jun · Dec`); every month → `All year`; nothing → ``.
 */
export function monthRange(months: number[]): string {
  const set = new Set(months);
  if (set.size === 0) return '';
  if (set.size === 12) return 'All year';
  const prev = (m: number) => ((m + 10) % 12) + 1;
  const next = (m: number) => (m % 12) + 1;
  const starts = [...set].filter((m) => !set.has(prev(m))).sort((a, b) => a - b);
  return starts
    .map((start) => {
      let end = start;
      while (set.has(next(end))) end = next(end);
      return start === end ? MONTHS[start - 1] : `${MONTHS[start - 1]} – ${MONTHS[end - 1]}`;
    })
    .join(' · ');
}

File: api/app/services/format.py
"""Money, dates and labels as the PDF and the emails print them — a mirror of
web/src/lib/format.ts (change both)."""

import datetime as dt

WEEKDAYS = ("Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun")
MONTHS = (
    "Jan",
    "Feb",
    "Mar",
    "Apr",
    "May",
    "Jun",
    "Jul",
    "Aug",
    "Sep",
    "Oct",
    "Nov",
    "Dec",
)


def inr(rupees: int) -> str:
    """Indian grouping: ₹12,34,567."""
    s = str(abs(rupees))
    if len(s) > 3:
        head, tail = s[:-3], s[-3:]
        groups = [head[max(i - 2, 0) : i] for i in range(len(head), 0, -2)][::-1]
        s = ",".join(groups) + "," + tail
    return ("-" if rupees < 0 else "") + "₹" + s


def long_date(d: dt.date) -> str:
    """`Fri 18 Dec 2026` — no zero padding, like the web's `formatDate`."""
    return f"{WEEKDAYS[d.weekday()]} {d.day} {MONTHS[d.month - 1]} {d.year}"


def duration(nights: int, days: int) -> str:
    """The web's `duration` — short form, everywhere on the site."""
    return f"{nights}N / {days}D"


def meals_label(breakfast: bool, lunch: bool, dinner: bool) -> str:
    """Format meals as `Breakfast · Lunch · Dinner` or `No meals`."""
    names = [
        n
        for n, on in (
            ("Breakfast", breakfast),
            ("Lunch", lunch),
            ("Dinner", dinner),
        )
        if on
    ]
    return " · ".join(names) if names else "No meals"


def seats_label(seats_left: int) -> str:
    """Format seat availability as `6 seats`, `1 seat`, or `Sold out`."""
    if seats_left <= 0:
        return "Sold out"
    return f"{seats_left} seat{'s' if seats_left != 1 else ''}"
